release/ray_release/test.py

```diff
@@ -414,6 +414,8 @@ def get_byod_runtime_env(self) -> Dict[str, str]:
             # Logs the full stack trace from Ray Data in case of exception,
             # which is useful for debugging failures.
             "RAY_DATA_LOG_INTERNAL_STACK_TRACE_TO_STDOUT": "1",
+            # To make ray data compatible across multiple pyarrow versions.
+            "RAY_DATA_AUTOLOAD_PYEXTENSIONTYPE": "1",
```

Comment:
CC: @scottjlee , @bveeramani - add this so the tests are compatible across multiple pyarrow versions